setup.py

```diff
@@ -75,12 +75,12 @@ class BuildExt(build_ext):
 
     c_opts = {
         "msvc": ["-EHsc", "-O2", "-W1", "-std:c++17", "-D_USE_MATH_DEFINES"],
-        "unix": ["-O3", "-W", "-fPIC", "-shared", "-fopenmp"],
+        "unix": ["-O3", "-W", "-Wno-reorder", "-fPIC", "-shared", "-fopenmp"],
```

Comment:
It may be better to reorder the constructor initializer lists rather than ignore this. 

Also, we should opt to enable all warnings as errors both here and in the CMake (see the [Jet repo](https://github.com/XanaduAI/jet/blob/cb4ab418cfae45a74c226ab354abb6a86e77825d/CMakeLists.txt#L205) for how we did it there).

Lastly, we need these in the Windows and MacOS flags too to keep it consistent (eventually I expect to move away from the build with `setup.py` to instead have it call CMake, but that will need to wait for now). The CI checks here should be enough to know if it works on there.

Reply:
well caught! my bad here. This was a temporary fix of variable reordering warnings.

Reply:
No worries :slightly_smiling_face: